File: sut.py
from tree import BPlusTree
import sys

tree = BPlusTree()

def handle_command(cmd):
    parts = cmd.strip().split()
    if not parts:
        return ""

    op = parts[0]

    if op == 'insert':
        for i in range(1, len(parts), 2):
            k = int(parts[i])
            v = int(parts[i+1])
            tree.insert(k, v)
        return "ok"

    elif op == 'remove':
        for k in parts[1:]:
            tree.remove(int(k))
        return "ok"

    elif op == 'lookup':
        r = tree.lookup(int(parts[1]))
        return str(r) if r is not None else "null"

    elif op == 'lowerbound':
        r = tree.lowerbound(int(parts[1]))
        return str(r) if r is not None else "null"

    return "unknown command"

if __name__ == "__main__":
    for line in sys.stdin:
        line = line.strip()
        if not line:
            continue
        print(handle_command(line))
        sys.stdout.flush()

File: sut.cpp
#include "header.h"
#include <sstream>

string handle_command(const string& command, BPlusTree& tree) {
    stringstream ss(command);
    string op;
    ss >> op;

    if (op == "insert") {
        int k; double v;
        while (ss >> k >> v)
            tree.insert(k, v);
        return "ok";
    } else if (op == "remove") {
        int k;
        while (ss >> k)
            tree.remove(k);
        return "ok";
    } else if (op == "lookup") {
        int k; ss >> k;
        double* val = tree.lookup(k);
        return val ? to_string(*val) : "null";
    } else if (op == "lowerbound") {
        int k; ss >> k;
        double* val = tree.lowerbound(k);
        return val ? to_string(*val) : "null";
    }
    return "unknown command";
}

int main() {
    BPlusTree tree(4);  // fixed degree 4
    string line;
    getline(cin, line);  // skip empty line or config
    while (getline(cin, line)) {
        if (line.empty()) continue;
        cout << handle_command(line, tree) << endl;
        cout.flush();
    }
    return 0;
}